pilot/pkg/kube/inject/webhook_test.go

```diff
@@ -966,6 +967,20 @@ func mergeParamsIntoHelmValues(params *Params, vals string, t testing.TB) string
 }
 
 func escapeHelmValue(val string) string {
+	if len(val) == 0 {
+		return val
+	}
+
+	if val[0] == '{' && val[len(val)-1] == '}' {
```

Comment:
minor point, can we just simply do 
```
val = strings.Replace(val, "{", "\\{", -1)
val = strings.Replace(val, "}", "\\}", -1)
```
or is there a need for the extra logic?

Reply:
We cannot because it will escape the outside two characters '{' and '}' which will be interpreted as string not array.

I think we need the logic to check if the input string is an array or a normal value. For array string, we need to escape `{` and `}` inside